File: src/main/java/gargoyle/netsquares/store/prop/PrefBooleanProperty.java
package gargoyle.netsquares.store.prop;

public class PrefBooleanProperty extends BasePrefProperty<Boolean> {
    public PrefBooleanProperty(Class<?> pkg, String key, Boolean defaultValue) {
        super(pkg, key, defaultValue);
    }

    @Override
    protected void _set(Boolean value) {
        preferences.putBoolean(key, value);
    }

    public Boolean get(Boolean defaultValue) {
        return preferences.getBoolean(key, defaultValue);
    }

    public void setValue(boolean value) {
        set(value);
    }
}

File: src/main/kotlin/gargoyle/netsquares/store/prop/PrefBooleanProperty.kt
package gargoyle.netsquares.store.prop

import kotlin.reflect.KClass

class PrefBooleanProperty(pkg: KClass<*>, key: String, defaultValue: Boolean) :
    BasePrefProperty<Boolean>(pkg, key, defaultValue) {
    override fun _set(value: Boolean) {
        preferences.putBoolean(key, value)
    }

    override fun get(defaultValue: Boolean): Boolean {
        return preferences.getBoolean(key, defaultValue)
    }

    fun setValue(value: Boolean) {
        set(value)
    }
}
